src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs

```diff
@@ -18,39 +16,38 @@ public UserType(IStringLocalizer<UserType> localizer)
         S = localizer;
 
         Name = "User";
-        Description = S["Represents the currently authenticated user."];
+        Description = S["Represents a user."];
 
         Field(u => u.UserId).Description(S["The id of the user."]);
         Field(u => u.UserName).Description(S["The name of the user."]);
         Field(u => u.Email, nullable: true).Description(S["The email of the user."]);
         Field(u => u.PhoneNumber, nullable: true).Description(S["The phone number of the user."]);
     }
 
-    // Adds a custom user settings field
-    internal void AddField(ISchema schema, ContentTypeDefinition typeDefinition)
+    public override void Initialize(ISchema schema)
     {
-        var contentItemType = schema.AdditionalTypeInstances.SingleOrDefault(t => t.Name == typeDefinition.Name);
-        
-        if (contentItemType == null)
+        // Add custom user settings by reusing previously registered content types with the
+        // stereotype "CustomUserSettings".
+        foreach (var contentItemType in schema.AdditionalTypeInstances.Where(t => t.Metadata.TryGetValue("Stereotype", out var stereotype) && stereotype as string == "CustomUserSettings"))
         {
-            // This error would indicate that this graph type is build too early.
-            throw new InvalidOperationException("ContentTypeDefinition has not been registered in GraphQL");
-        }
-
-        var field = Field(typeDefinition.Name, contentItemType.GetType())
-            .Description(S["Custom user settings of {0}.", typeDefinition.DisplayName])
-            .ResolveAsync(static async context => {
-                // We don't want to create an empty content item if it does not exist.
-                if (context.Source is User user &&
-                    user.Properties.ContainsKey(context.FieldDefinition.ResolvedType.Name))
+            Field(contentItemType.Name, contentItemType)
+                .Description(S["Custom user settings of {0}.", contentItemType.Name])
+                .ResolveAsync(static async context =>
                 {
-                    var customUserSettingsService = context.RequestServices!.GetRequiredService<CustomUserSettingsService>();
-                    var settingsType = await customUserSettingsService.GetSettingsTypeAsync(context.FieldDefinition.ResolvedType.Name);
-
-                    return await customUserSettingsService.GetSettingsAsync(user, settingsType);
-                }
+                    // We don't want to create an empty content item if it does not exist.
+                    if (context.Source is User user &&
+                        user.Properties.ContainsKey(context.FieldDefinition.ResolvedType.Name))
+                    {
+                        var customUserSettingsService = context.RequestServices!.GetRequiredService<CustomUserSettingsService>();
```

Comment:
Why `RequestServices!`

Reply:
It's defined as nullable, but can't be in our case as far as I know. If you don't want nullable annotations being used, I can of cause remove them.

Reply:
Okay, I removed it. But just for completeness, it was already used before this PR 🤷‍♂️

Reply:
I see, just leave my comment